operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/IndexDiscriminator.java

```diff
@@ -0,0 +1,45 @@
+package io.javaoperatorsdk.operator.api.reconciler;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+
+public class IndexDiscriminator<R extends HasMetadata, P extends HasMetadata>
```

Comment:
Some documentation to explain what this does and why it's useful would be helpful for users.

Reply:
added